### DISCUSSION TITLE
How to trigger on user accept email invite?

### DISCUSSION THREAD

**Discussion by dilly-sixmo** [supabase/supabase#10613] [Questions] [answered]
I have a table that lists all the users of my system.  I would like to display whether or not each user has accepted their email invitation.  As that information is stored in the `auth.users` table, which is inaccessible, I need to come up with some other way to determine this.

I already have a trigger that creates a new row in my `profiles` table when a new `user` signs up.  I was thinking that I might be able to use another trigger to set a `signup_complete` field in the `profiles` table to true when a user accepts their invite, but I don't know how to accomplish that.

Ideally, I would trigger a function when the `email_confirmed_at` field in `auth.users` gets updated, but based on what I'm seeing, you can't execute a trigger on a specific column update.

Does anyone know how I can accomplish this?  I just need a way to determine if a user has accepted their email invite.

**Comment by GaryAustin1** [comment, marked as answer]
You have a trigger on update and can check if new.email_confirmed_at versus old.email_confirmed_at are different.  Or possibly just check if old.email_confirmed_at is null and you have a time in new.email_confirmed_at.

**Reply by dilly-sixmo** [reply]
Ok.  So I could determine what field was updated within the function itself.  I was looking for that option in the "create trigger" dialog.  And apparently the database functions have `new` and `old` variables so I can access the old data before modification?  I was not aware of that.  I think that will allow me to get this to work.  Thanks!

**Reply by GaryAustin1** [reply]
https://www.postgresql.org/docs/current/plpgsql-trigger.html
More info on what all is in a trigger function.
